backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +56,49 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(ignore_result=False, bind=True, queue="migrator")
+def recalculate_references_for_merged_record(self, merged_record_uri, record_uri):
+    merged_record_schema = merged_record_uri.rsplit("/", 2)[1]
+    if merged_record_schema == "literature":
+        merged_record_schema = "hep"
+    referenced_records_indices_and_paths = get_refs_to_schemas()[merged_record_schema]
+    match_reference_with_config(
+        referenced_records_indices_and_paths, merged_record_uri, record_uri
+    )
+
+
+def match_reference_with_config(config_list, merged_record_uri, new_record_uri):
+    for index, path in config_list:
+        config = (
+            generate_matcher_config_for_nested_reference_field(index, path)
+            if path in ["authors", "publication_info", "first_author", "supervisors"]
```

Comment:
* Don't hide the list of `nested_paths` here. It would be better to add a `nested_record_fields` `property` to `InspireRecord` (empty by default) and override it for `LiteratureRecord`. That would also prevent name collisions between nested and non-nested fields in different indices (we don't have those currently).
* You don't need to search in `first_author`, as it's a subset of `authors` used only for searching. It's not present in the record, so there's nothing to update.